src/dependency/Pyegi/Pyegi/installer.py

```diff
@@ -0,0 +1,169 @@
+                    os.symlink(src, dst)
+            else:
+                packages_to_move.append(name_in_commons)
+        phase1_end_time = dt1.now()
+        print("Total phase 1 time: " + str(phase1_end_time - phase1_start_time))
+
+        os.system("poetry install")
+        phase2_start_time = dt1.now()
+        for name_in_commons in packages_to_move:
+            filename = (
+                f"{temp_dir}.venv/Lib/site-packages/{name_in_commons}.dist-info/RECORD"
+            )
+            if exists(filename):
+                paths = []
+                with open(filename, "r") as csvfile:
+                    csvreader = csv.reader(csvfile)
+                    for row in csvreader:
+                        paths.append(row[0])
+                for path in paths:
+                    connection_path = "/Lib/site-packages/"
+                    if path[:3] == "../":
+                        connection_path = "/Lib/"
+                        path = path[3:]
+                        if path[:3] == "../":
+                            connection_path = "/"
+                            path = path[3:]
+                    src = f"{temp_dir}.venv{connection_path + path}"
+                    dst = commons_dir + name_in_commons + connection_path + path
+                    create_dirs(dst)
+                    shutil.move(src, dst)
+                    dst2 = f"{script_path}.venv{connection_path + path}"
+                    create_dirs(dst2)
+                    os.symlink(dst, dst2)
+        phase2_end_time = dt1.now()
+        print("Total phase 2 time: " + str(phase2_end_time - phase2_start_time))
+        os.chdir(os.path.dirname(__file__))
+        shutil.rmtree(temp_dir)
+        os.mkdir("temp")
+        toml_file = open(temp_dir + "poetry.toml", "w")
+        toml_file.write("[virtualenvs]\nin-project = true")
+        toml_file.close()
+    else:
+        print(f'The file "pyproject.toml" doesn\'t exist in {script} script directory.')
+
+
+def clean_lib_links(script):
+    f = open(commons_dir + "lib_links.json")
+    lib_links = json.load(f)
+    f.close()
+    pkgs = []
+    zero_pkgs = []
+    for package in lib_links["Packages"]:
+        if script in package["scripts"]:
+            package["scripts"].remove(script)
+            if len(package["scripts"]) == 0:
+                if exists(commons_dir + package["name"]):
+                    zero_pkgs.append(package["name"])
+            else:
+                pkgs.append(package)
+        else:
+            pkgs.append(package)
+    lib_links["Packages"] = pkgs
+    json.dump(lib_links, open(commons_dir + "lib_links.json", "w"))
+    return zero_pkgs
+
+
+def uninstall_pkg(script):
+    print(f"Processing {script} dependencies...")
+    script_path = scriptsPath + script + "/"
+    shutil.rmtree(script_path)
+    zero_pkgs = clean_lib_links(script)
+    for zero_pkg in zero_pkgs:
+        shutil.rmtree(commons_dir + zero_pkg)
+
+
+for script in scripts_names:
```

Comment:
```
if __name__ == "__main__":
    scripts_names = [
        name
        for name in os.listdir(scriptsPath)
        if os.path.isdir(os.path.join(scriptsPath, name))
    ]
    for script in scripts_names:
        install_pkg(script)
    # install_pkg("[sample] Disintegration")
```

the `scripts_names` in top of the file should get deleted too.

Reply:
will this help with importing functions from this file to other files? or are we going to do all the install/uninstall/upgrade within this file?

Reply:
it helps to import functions in other files.
for now, I don't know what we will do. but we should prevent causing artifacts.